main.go

```diff
@@ -10,21 +10,39 @@ import (
 )
 
 type ProxyConfig struct {
+	// Port on whch the proxy is listening on
 	Port int `env:"PORT" envDefault:"3000"`
 
 	// Proxy target
 	RemoteAddress string `env:"REMOTE_ADDRESS"`
 
-	// InputValidation enables the proxy to validate the incoming proxy
-	// requests using
+	// InputValidation enables the proxy to validate / authenticate the
+	// incoming proxy requests using
 	InputValidation bool `env:"INPUT_VALIDATION"`
-	// Allowed Public identities passed in as comma separated b64 encoded
-	// pubkeys
+
+	// one of: [ "ecdsasignatures", "basicauth", "none" ]
+	AuthenticationScheme string `env:"AUTHENTICATION_SCHEME" envDefault:"basicauth"`
+
+	// if AuthenticationScheme is basicauth then this string is used for
+	// determinig allowed user / password pairs.
+	// NOTE: the string format is: user1:password1,user2:password2
+	AllowedBasicAuthUserString string `env:"ALLOWED_USERS_STRING" envDefault:""`
```

Comment:
Let's keep the name the env string in-sync. I feel like you're trying to make the env strings shorter but then you also lose information. E.g., in this case that these user strings are used for basic auth.

Reply:
👍